### PR TITLE
Add EdgeCore ECS4110-28T Support

### PR DESCRIPTION
**Pull request by soto2080**
* Since I can't find MIBs anywhere, I decide to use numeric oids till I find it.
* Just add EdgeCos recognition and some hardware infos.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by soto2080** [comment]
Mmm so the commits are OK?
Or I need to provide more info?

**Comment by PipoCanaja** [comment]
Hi @soto2080 
You need to run again all tests from edgecos manually on your dev machine, and add them to the PR, to fix the lowerlayerdown status. If you look at Travis results, you get the reason of the failure on 'edgecos_es3510ma' but essentially all edgecos tests that have portStatus would have the same issue.

**Comment by PipoCanaja** [comment]
Hi @soto2080 
You still need to re-run the existing tests that are impacted by this change. TravisCI gives you the steps where they fail, so you know which one you have to re-run. 
You can also use the pre-commit script to help you find out. 
Bye

**Comment by soto2080** [comment]
I encounter a problem that save-test-data.php still crash on my machine.
Is there docs about environment variables or python dependency?
I am on CentOS7.

**Comment by soto2080** [comment]
I may have time to look into how to make the test data.

File: includes/polling/ports/edgecos.inc.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+* @package    LibreNMS
+* @link       http://librenms.org
+* @copyright  2019 Ming-Han Yang
+* @author     Ming-Han Yang <soto2080edu@g.ncu.edu.tw>
+*/
+foreach ($port_stats as & $port) {
+    if ($port['ifOperStatus'] == 'lowerLayerDown') {
```
Comment (murrant) [inline includes/polling/ports/edgecos.inc.php]: Why did you change this?  lowerLayerDown is a proper ifOperStatus...
If you are having issues with a rule, you can just make it check ifOperStatus = down or ifOperStatus = lowerLayerDown

Comment (soto2080) [reply includes/polling/ports/edgecos.inc.php]: The problem is that although lowerLayerDown is defined in the MIB, but the biggest players do not follow the defines i.e Cisco always make ifOperStatus as 'down' even it should be lowerLayerDown or others in the status.

Btw, another problem is that the whole front end of the web don't have the logic that there are many type of admin up but port down.

The web front just check if port is up/up, if yes, all fine.
but if it has admin down, if yes, assign as admin down,
If not, then the problem occur, only ifOperStatus=='down' will be seem as down, all the other ifOperStatus will not be anyone of up/down/admindown, it just disappear in the frontend.

Comment (murrant) [reply includes/polling/ports/edgecos.inc.php]: IMHO, changing the data is not the correct way to resolve that.     Checks should probably be ifOperStatus == 'up' or ifOperStatus != 'up'.

Possible ifOperStatus status:
1: up(1)
2: down(2)
3: testing(3)
4: unknown(4)
5: dormant(5)
6: notPresent(6)
7: lowerLayerDown(7)

Possible ifAdminStatus:
1: up(1)    
2: down(2)
3: testing(3)

Comment (soto2080) [reply includes/polling/ports/edgecos.inc.php]: I agree with you but there is no way to show these info in the frontend.

**Comment by soto2080** [comment]
If these commits are ok, please merge it.
The port status issue I will look into it later for a more elegant fix.
But as far as I know, the logic is totally coupling with $port['ifOperStatus'] === 'down' .

**Comment by murrant** [comment]
Yep, easy merge :+1:

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-56-release-changelog-september-2019/9671/1
